contracts/cyberpunk/src/contract.rs

```diff
@@ -179,25 +179,51 @@ fn execute_debug(api: &dyn Api) -> Result<Response, ContractError> {
 }
 
 #[entry_point]
-pub fn query(_deps: Deps, env: Env, msg: QueryMsg) -> StdResult<QueryResponse> {
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<QueryResponse> {
     use QueryMsg::*;
 
     match msg {
         MirrorEnv {} => to_binary(&query_mirror_env(env)),
+        Denoms {} => to_binary(&query_denoms(deps)?),
     }
 }
 
 fn query_mirror_env(env: Env) -> Env {
     env
 }
 
+fn query_denoms(deps: Deps) -> StdResult<(String, Vec<String>)> {
```

Comment:
It would be cool to get the whole `DenomMetadata` structs as elements for the testing